src/v/cluster/controller.h

```diff
@@ -151,11 +151,23 @@ class controller {
 private:
     friend controller_probe;
 
+    enum class create_cluster_mode { bootstrap, upgrade };
```

Comment:
While it's true this is only relevant in upgrade scenarios, I think a casual glance through this naming will mislead people into thinking this is relevant for all upgrades to any version. Could we call this `cluster_uuid_only` instead?

Reply:
I just wanted to abstract that from the actual actions performed. Maybe `cluster_upgrade` is clearer?

Reply:
I don't think that's clearer because it is still specific only to upgrades from before v22.2.

IMO it'd be clearer what's actually being done when we say `cluster_uuid_only` -- only create a cluster UUID, not anything else